modules/qna/src/backend/api.ts

```diff
@@ -130,6 +139,15 @@ export default async (bp: typeof sdk, botScopedStorage: Map<string, QnaStorage>)
 
       const statusCb = processedCount =>
         updateUploadStatus(uploadStatusId, `Deleted ${processedCount}/${questions.length} questions`)
+
+      //Delete Content Elements
+      for (const question of questions) {
+        const axiosConfig = await bp.http.getAxiosConfigForBot(req.params.botId);
+        await axios.post(`/content/elements/bulk_delete?flag=qna`,
+          question.data.answers.map(a => a.contentId).filter(a => a != undefined),
+          { ...axiosConfig, headers: { authorization: req.headers.authorization } })
+      }
+
```

Comment:
@slvnperron  Can you guys tell me why the token from getAxiosConfigForBot doesn't have permission to delete contentElements? this is the reason why this is here by the way :D

Reply:
@allardy That, and I think we should put delete straight in the SDK?

Reply:
@slvnperron @davidvitora I've added the delete content element method in the sdk for next release.

I think we are addressing this feature the wrong way around. The best solution would be to have all the QNA answers in the CMS, and let the user pick a content element to answer the question. This would allow the bot to support multiple languages, and there would only be one way to do things.

In the short term, I think we could simply add the BotpressContentPicker widget,, since it handles the selection & creation of new elements in the same page (there is an example in the basic-skills  module).

What do you guys think?

Reply:
@allardy In order for that to work we need to handle contentElement deletion in the way that @slvnperron mentioned

> @davidvitora hey thanks so much for the awesome PR!
> 
> I just have question for the rest of the team w.r.t where the content elements are stored.
> 
> @allardy @EFF @rndlaine here the elements are stored directly inside the QnA files instead of in the `/content-elements` like the rest of the content.
> 
> My thinking is that all content-elements without exception should be in the same location but I'm wondering how we handle deletion of content from outside the QnA? Should it disable the QnA pair if there's no more answers inside the array?
> 
> Also I think the modules could expose a handler to handle CMS elements add/deletion/edits? Similar to how we do when we rename flows:
> [botpress/src/bp/sdk/botpress.d.ts](https://github.com/botpress/botpress/blob/fe7806fca7a2607c61cb29071c913c25893f96dc/src/bp/sdk/botpress.d.ts#L87-L88)
> 
> Lines 87 to 88 in [fe7806f](/botpress/botpress/commit/fe7806fca7a2607c61cb29071c913c25893f96dc)
> 
>    onFlowChanged?: ((bp: typeof import('botpress/sdk'), botId: string, flow: Flow) => void) 
>  }

Reply:
@davidvitora You are right, I missed that part of the discussion. 

I suggest adding an `onContentElementChanged` property that modules can define, which will be called with parameters `botId`, `contentElement` (the concerned object) and  `action` (add, edit, delete). 

What do you think?

Reply:
@allardy i think that it is perfect, you guys could also use that in the flow builder, since today it has the same issue when contentElements are deleted.

I'm also thinking about another issue that might happen, i think that this is a bigger problem that already happens in Botpress, please let me know if you have something in mind for it: 

We have about 150+ questions for each bot saved in a CSV file, we are doing some tests before putting bots in production using a branch that i have created internally which have my 3 PR's (1549, 1577 and 1647). Today when i import questions using the CSV file, it creates the contentElement of each answer. But lets say that we need to import it again, so previous questions must be deleted, but in order to not have the contentManager full of not used contentElements, we need to delete them, so in my branch with 1549(this), since every question creates his own contentElement, the module just deletes all contentElements of each question.

In brief, there is several operations that can happen in which we need to clean the contentManager, eg: The answer with contentElement was removed from a question or the question was removed in order to re-import questions using CSV.

The contentElement can only be deleted if it is not being used in other locations, eg: flows or other questions from QNA.

I have some ideas in mind, but i would like to know your opinion about this.

Reply:
Each module could implement a function to check if a content element is used within itself, so from the element we could call a function to check if it is used in somewhere else, and if so, the module would behave behave differently when deleting the element, just removing his link to the contentElement

Reply:
@allardy this would remove the need to use flags, that's because the contentManager would check if the contentElement is used somewhere else when trying to delete it. If we also implement onContentElementChanged, we could warn the user that the element is being used instead of just blocking the deletion

Reply:
@davidvitora I've opened this PR to add the feature: https://github.com/botpress/botpress/pull/1696

For now, it is triggered when an element is created, updated or deleted, but there is no way to cancel the action, this would require more work on the UI and backend beforehand. 

Do you think this will be good for now?
Thanks

Reply:
@allardy Yes i think it will be very usefull, I will put that in use as soon as you merge it, we can plan to fix the other issue in the future PRs. Currently I'm modifying this branch to behave the way you requested, did you know that the json file from the content type doesn't get updated when a element is deleted? Is that intended?

Reply:
@davidvitora Absolutely not, good catch!  It will be added in that PR also. 

With the addition of multi-language support, your feature is a lot more useful, since content must be in the CMS to be translated. I think adding the content picker to the QNA would be good, and displaying the preview of the element in the "Answer" field of the question.

I'm not sure of what would be the best UX for this tho.

Reply:
@allardy yes i agree that it has to use the contentPicker, i will be pushing the changes soon